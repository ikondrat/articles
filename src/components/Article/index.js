import React from 'react';
import {Card, CardContent} from '@material-ui/core';
import PropTypes from 'prop-types';
import {
  Link
} from 'react-router-dom'
import {
  AppBar,
  Toolbar,
  Divider
} from '@material-ui/core';
import styled from 'styled-components';
import {
  Progress
} from '../../components';

const StyledBar = styled(AppBar)`
  && {
    position: relative;
    a {
      color: white;
    }
  }
`;

const Article = ({
  title,
  url,
  isFetching
}) => (
  <div>
    <StyledBar>
        <Toolbar>
            <Link to="/">Go back</Link>
        </Toolbar>
    </StyledBar>
    <Divider/>
    {isFetching &&
      <Progress/>
    }
    {!isFetching &&
      <Card>
        <CardContent>
          <h1>{title}</h1>
          <a href={url}>{url}</a>
        </CardContent>
      </Card>
    }
  </div>
);

Article.propTypes = {
  title: PropTypes.string,
  url: PropTypes.string,
  isFetching: PropTypes.bool
};


export default Article;