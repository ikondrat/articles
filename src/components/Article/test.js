import React from 'react';
import { shallow } from 'enzyme';
import Article from './';
import '../../setupTests';

it('renders without crashing', () => {
  shallow(<Article 
    title="test title"
  />);
});