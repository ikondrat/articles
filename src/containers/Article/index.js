import React, { Component } from 'react'
import { connect } from 'react-redux'
import { 
  fetchArticle
} from '../../actions'
import Article from '../../components/Article'

class ArtilceContainer extends Component {
  componentWillMount() {
    this.props.fetchArticle(this.props.match.params.articleId)
  }
  render() {
    return (<Article {...this.props}/>)
  }
}

function mapStateToProps(state) {
  return {
    ...state.article.articleDetails,
    isFetching: state.article.isFetching
  };
}

const bindActionsToDispatch = dispatch => ({
  fetchArticle : (x) => {dispatch(fetchArticle(x))}
});

export default connect(mapStateToProps, bindActionsToDispatch)(ArtilceContainer)