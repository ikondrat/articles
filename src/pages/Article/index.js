import React from 'react';
import Article from '../../containers/Article';

export default (match) => (
    <div>
        <Article {...match}/>
    </div>
)