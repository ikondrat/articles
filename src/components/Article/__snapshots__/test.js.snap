// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Article
    isFetching={true}
    title="test title"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Styled(WithStyles(AppBar))>
          <WithStyles(Toolbar)>
                    <Link
                              replace={false}
                              to="/"
                    >
                              Go back
                    </Link>
          </WithStyles(Toolbar)>
</Styled(WithStyles(AppBar))>,
        <WithStyles(Divider) />,
        <Unknown />,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Toolbar)>
            <Link
                        replace={false}
                        to="/"
            >
                        Go back
            </Link>
</WithStyles(Toolbar)>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Link
              replace={false}
              to="/"
>
              Go back
</Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Go back",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": "Go back",
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Styled(WithStyles(AppBar))>
            <WithStyles(Toolbar)>
                        <Link
                                    replace={false}
                                    to="/"
                        >
                                    Go back
                        </Link>
            </WithStyles(Toolbar)>
</Styled(WithStyles(AppBar))>,
          <WithStyles(Divider) />,
          <Unknown />,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Toolbar)>
              <Link
                            replace={false}
                            to="/"
              >
                            Go back
              </Link>
</WithStyles(Toolbar)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Link
                replace={false}
                to="/"
>
                Go back
</Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Go back",
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": "Go back",
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`renders correctly 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Article
    isFetching={false}
    title="test title"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Styled(WithStyles(AppBar))>
          <WithStyles(Toolbar)>
                    <Link
                              replace={false}
                              to="/"
                    >
                              Go back
                    </Link>
          </WithStyles(Toolbar)>
</Styled(WithStyles(AppBar))>,
        <WithStyles(Divider) />,
        false,
        <WithStyles(Card)>
          <WithStyles(CardContent)>
                    <h1>
                              test title
                    </h1>
                    <a
                              href={undefined}
                    />
          </WithStyles(CardContent)>
</WithStyles(Card)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Toolbar)>
            <Link
                        replace={false}
                        to="/"
            >
                        Go back
            </Link>
</WithStyles(Toolbar)>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Link
              replace={false}
              to="/"
>
              Go back
</Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Go back",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": "Go back",
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(CardContent)>
            <h1>
                        test title
            </h1>
            <a
                        href={undefined}
            />
</WithStyles(CardContent)>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <h1>
                test title
</h1>,
              <a
                href={undefined}
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "test title",
              },
              "ref": null,
              "rendered": "test title",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "href": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "a",
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Styled(WithStyles(AppBar))>
            <WithStyles(Toolbar)>
                        <Link
                                    replace={false}
                                    to="/"
                        >
                                    Go back
                        </Link>
            </WithStyles(Toolbar)>
</Styled(WithStyles(AppBar))>,
          <WithStyles(Divider) />,
          false,
          <WithStyles(Card)>
            <WithStyles(CardContent)>
                        <h1>
                                    test title
                        </h1>
                        <a
                                    href={undefined}
                        />
            </WithStyles(CardContent)>
</WithStyles(Card)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Toolbar)>
              <Link
                            replace={false}
                            to="/"
              >
                            Go back
              </Link>
</WithStyles(Toolbar)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Link
                replace={false}
                to="/"
>
                Go back
</Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Go back",
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": "Go back",
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(CardContent)>
              <h1>
                            test title
              </h1>
              <a
                            href={undefined}
              />
</WithStyles(CardContent)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <h1>
                  test title
</h1>,
                <a
                  href={undefined}
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "test title",
                },
                "ref": null,
                "rendered": "test title",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "href": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "a",
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
